packages/aws-cdk-lib/aws-lambda/test/function.test.ts

```diff
@@ -3832,6 +3832,58 @@ describe('VPC configuration', () => {
       allowAllOutbound: false,
     })).not.toThrow();
   });
+
+  test('with ipv6AllowedForDualStack and no VPC', () => {
+    const stack = new cdk.Stack();
+    expect(() => new lambda.Function(stack, 'MyLambda', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.PYTHON_3_9,
+      ipv6AllowedForDualStack: true,
+    })).toThrow(/Cannot configure 'ipv6AllowedForDualStack' without configuring a VPC/);
+  });
+
+  test('set ipv6AllowedForDualStack with VPC', () => {
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'Vpc', {
+      maxAzs: 3,
+      natGateways: 1,
+    });
+    const securityGroup = new ec2.SecurityGroup(stack, 'LambdaSG', {
+      vpc,
+      allowAllOutbound: true,
+      allowAllIpv6Outbound: true,
+    });
+    new lambda.Function(stack, 'MyLambda', {
+      vpc: vpc,
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.PYTHON_3_9,
+      ipv6AllowedForDualStack: true,
```

Comment:
Could we add a tiny test with ipv6AllowedForDualStack as false to make sure it works as expect?

Reply:
Added new test to check when `ipv6AllowedForDualStack` is set to false